### ISSUE TITLE
assignment operation inside an 'if' condition should be marked with a warning

### ISSUE THREAD

**Issue by AGiorgetti** [palantir/tslint#507]
If we want to spot some more bugs at compile time, I believe that something like:

```javascript
if (data.id = id) { ...
```

while being perfectly legal, it can be a coding bug most of the times.
it would be nice if TsLint could spot this type of code and give us a warning instead of using a custom regex that scan all our files.

**Comment by adidahiya** [comment]
sounds good. it could be called... `no-assignment-in-conditional`? cc @ashwinr for thoughts on rule name

**Comment by ashwinr** [comment]
eslint has the same [rule](http://eslint.org/docs/rules/no-cond-assign). even just `no-conditional-assignment` seems good.

**Comment by JKillian** [comment]
ESLint has an option for the rule where they allow assignments that are wrapped in parenthesis. Do we want the same option or should we ust flag all assignments in a conditional no matter what?

**Comment by adidahiya** [comment]
let's just do the simple thing for now and disallow all assignments in a conditional. we can add the other option later if it's requested.
